### PR TITLE
Change Default for `psatd.current_correction`

### PR DESCRIPTION
**Pull request by EZoni**
Change default for `psatd.current_correction`: always active, unless either a charge-conserving current deposition scheme (Esirkepov, Vay) or the div(E) scheme are used. 

Also, due to the new default, we do not require anymore that the current correction be used only in combination with the periodic single box option.

To-do:
- [x] Fix CI tests (will deactivate current correction to keep CI status as it was before this change)
- [x] Add warnings (if charge conservation is not guaranteed by either of the available methods)

P.S. An example of a realistic test case where charge conservation plays a significant role is given in #2462.

### CODE REVIEW

File: Source/WarpX.cpp [hunk 1]
```diff
@@ -1303,11 +1306,6 @@ WarpX::ReadParameters ()
             "Vay current deposition not implemented for Galilean algorithms"
         );
 
-        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
-            (!current_correction) || v_galilean_is_zero || (!fft_do_time_averaging),
-            "Current correction not implemented for averaged Galilean algorithm"
-        );
```
Comment (EZoni) [inline Source/WarpX.cpp:1310]: I don't see obvious reasons why we should not be allowed to use the Galilean current correction with time averaging. Maybe to be discussed offline.

File: Source/WarpX.cpp [hunk 2]
```diff
@@ -1204,23 +1204,41 @@ WarpX::ReadParameters ()
...
+        // current deposition (Esirkepov, Vay) or the div(E) cleaning scheme
+        // are used
+        current_correction = true;
+        if (WarpX::current_deposition_algo == CurrentDepositionAlgo::Esirkepov ||
+            WarpX::current_deposition_algo == CurrentDepositionAlgo::Vay ||
+            WarpX::do_dive_cleaning)
+        {
+            current_correction = false;
+        }
+
```
Comment (RemiLehe) [inline Source/WarpX.cpp:1223]: Could you confirm that the current_correction is only activated when we use a spectral (PSATD) Maxwell solver?
Also: could you mention the default value of `current_correction` in the documentation for this parameter?

File: Source/WarpX.cpp (hunk 1 above)
Comment (RemiLehe) [reply Source/WarpX.cpp:1310]: Hm, I agree we could remove this.

I seems that in general, the current correction is not implemented for the case where `J` is linear in time:
https://github.com/ECP-WarpX/WarpX/blob/development/Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithmJLinearInTime.cpp#L427
But we already have an abort there, so no need to have another check in `WarpX.cpp`.

Discussed offline, let's add an `amrex::Abort` in Source/Evolve/WarpXEvolve.cpp, inside the `OneStep_multiJ` function.

**Review by RemiLehe — CHANGES_REQUESTED** [review]

File: Source/WarpX.cpp (hunk 2 above)
Comment (EZoni) [reply Source/WarpX.cpp:1223]: Documentation updated in https://github.com/ECP-WarpX/WarpX/pull/3092/commits/cfd316c91bc66e22fe0dbba948dc91e442b5adae. The parameter itself is set with the `psatd` prefix, FDTD simulations will not parse it.

File: Source/WarpX.cpp (hunk 1 above)
Comment (EZoni) [reply Source/WarpX.cpp:1310]: Implemented in https://github.com/ECP-WarpX/WarpX/pull/3092/commits/ce22f93a6dd437a1daaede9dee2afd3d69fdabba (also, default changed to `False` for multi-J).

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
Let me add bug labels: "bad" defaults giving buggy/subideal physics (just so we don't overlook it).

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**